tcl/sql.tcl

```diff
@@ -62,11 +62,27 @@ proc qc::sql_sort { args } {
     }
     set order_by_list {}
     for {set i 0} {$i<[llength $list]} {incr i} {
+
 	set this_item [lindex $list $i]
         set this_item [string trim $this_item]
-        set this_item [qc::db_quote_identifier $this_item]
+        set parts [split $this_item "."]
+        switch [llength $parts] {
+            1 {
+                set this_item [qc::db_quote_identifier $this_item]                
+            }
+            2 {
+                set table [qc::db_quote_identifier [lindex $parts 0]]
+                set column [qc::db_quote_identifier [lindex $parts 1]]
+                set this_item "${table}.${column}"
+            }
+            default {
+                error "Unable to parse identifier \"$this_item\""
+            }
+        }
+
 	set next_item [lindex $list [expr {$i+1}]]
```

Comment:
looks like they are allowed but only within quoted identifiers.

Could we put the responsibility of quoting on the developer for columns
that contain ".".

Then only call db_quote_identifier if it is not already quoted

On 4 August 2016 at 15:44, Daniel Clark <daniel@qcode.co.uk> wrote:

> are "." allowed in postgresql identifiers ... checking docs ...
> https://www.postgresql.org/docs/9.1/static/sql-syntax-lexical.html
>
> On 4 August 2016 at 15:43, Bernhard van Woerden <notifications@github.com>
> wrote:
>
>> In tcl/sql.tcl
>> <https://github.com/qcode-software/qcode-tcl/pull/408#discussion_r73535474>
>> :
>>
>> > -        set this_item [qc::db_quote_identifier $this_item]
>> > +        set parts [split $this_item "."]
>> > +        switch [llength $parts] {
>> > +            1 {
>> > +                set this_item [qc::db_quote_identifier $this_item]
>> > +            }
>> > +            2 {
>> > +                set table [qc::db_quote_identifier [lindex $parts 0]]
>> > +                set column [qc::db_quote_identifier [lindex $parts 1]]
>> > +                set this_item "${table}.${column}"
>> > +            }
>> > +            default {
>> > +                error "Unable to parse identifier \"$this_item\""
>> > +            }
>> > +        }
>> > +
>> >  	set next_item [lindex $list [expr {$i+1}]]
>>
>> say we have a table "foo" with column "bar.baz"
>> correct identifier would be
>> "foo"."bar.baz" ?
>>
>> —
>> You are receiving this because you were assigned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/qcode-software/qcode-tcl/pull/408/files/2033187dd55e598a72e791a80cf4529ea2057dce#r73535474>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AC8lGY559nxqyh2YtxXCGMmmE-SPDANrks5qcfqZgaJpZM4Jcvnx>
>> .
>>
>
>
>
> --
> Kind Regards,
> Daniel Clark
> Qcode Software Limited
> www.qcode.co.uk
> Tel: +44(0) 1463 896 482
>



-- 
Kind Regards,
Daniel Clark
Qcode Software Limited
www.qcode.co.uk
Tel: +44(0) 1463 896 482